server/protocols/webadmin/webadmin-core/src/test/java/org/apache/james/webadmin/routes/HealthCheckRoutesTest.java

```diff
@@ -131,4 +139,100 @@ public void validateHealthchecksShouldReturnInternalErrorWhenOneHealthCheckIsDeg
         .then()
             .statusCode(HttpStatus.INTERNAL_SERVER_ERROR_500);
     }
+    
+    @Test
+    public void performHealthCheckShouldReturnOkWhenHealthCheckIsHealthy() {
+        healthChecks.add(healthCheck(Result.healthy(COMPONENT_NAME_1)));
+        
+        given()
+            .pathParam("componentName", COMPONENT_NAME_1.getName())
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .body("componentName", equalTo("component-1"))
+            .body("escapedComponentName", equalTo("component-1"))
+            .body("status", equalTo(ResultStatus.HEALTHY.getValue()))
+            .body("cause", is(nullValue()));
+    }
+    
+    @Test
+    public void performHealthCheckShouldReturnNotFoundWhenComponentNameIsUnknown() {
+        
+        given()
+            .pathParam("componentName", "unknown")
+        .when()
+            .get("/checks/{componentName}")
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404)
+            .body("details", is(nullValue()))
+            .body("type", equalTo("notFound"))
+            .body("message", equalTo("Component with name unknown cannot be found"))
+            .body("statusCode", is(404));
```

Comment:
`404` could be replaced by `HttpStatus.NOT_FOUND_404`

Reply:
fixed in 3dd0eacd799e503633a696fb238e5b0354d956b8